src/service.rs

```diff
@@ -110,27 +117,33 @@ impl<Types: NodeType> GlobalState<Types> {
             response_receiver,
             tx_sender,
             instance_state,
+            max_api_waiting_time,
         }
     }
 
     // remove the builder state handles based on the decide event
     pub fn remove_handles(
         &mut self,
         builder_vid_commitment: &VidCommitment,
-        block_hashes: Vec<BuilderCommitment>,
+        block_hashes: HashSet<(Types::Time, BuilderCommitment)>,
```

Comment:
This does not seem like the right container for this. Are we thinking we might have a (ViewN, BuilderCommitment) and a (ViewN+1, BuilderCommitment) where the BuilderCommitment is the same?

Reply:
yes,